lib/ansible/modules/storage/hpe3par/cpg_3par.py

```diff
@@ -0,0 +1,397 @@
+        state: present
+        cpg_name: sample_cpg
+        domain: sample_domain
+        growth_increment: 32000 MiB
+        growth_limit: 64000 MiB
+        growth_warning: 48000 MiB
+        raid_type: R6
+        set_size: 8
+        high_availability: MAG
+        disk_type: FC
+        secure: no
+
+    - name: Modify CPG sample_cpg
+      cpg_3par:
+        storage_system_ip: 10.10.10.1
+        storage_system_username: username
+        storage_system_password: password
+        state: present
+        cpg_name: sample_cpg
+        growth_increment: 36000 MiB
+        growth_limit: 65002 MiB
+        growth_warning: 45000 MiB
+        raid_type: R6
+        set_size: 8
+        high_availability: MAG
+        disk_type: FC
+        secure: no
+        new_name: new_sample_cpg
+        disable_auto_grow: false
+        rm_growth_limit: false
+        rm_warning_alert: false
+
+    - name: Delete CPG sample_cpg
+      cpg_3par:
+        storage_system_ip: 10.10.10.1
+        storage_system_username: username
+        storage_system_password: password
+        state: absent
+        cpg_name: sample_cpg
+        secure: no
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.storage.hpe3par import hpe3par
+try:
+    from hpe3par_sdk import client
+    from hpe3parclient import exceptions
+    HAS_3PARCLIENT = True
+except ImportError:
+    HAS_3PARCLIENT = False
+
+
+def validate_set_size(raid_type, set_size):
+    if raid_type:
+        set_size_array = client.HPE3ParClient.RAID_MAP[raid_type]['set_sizes']
+        if set_size in set_size_array:
+            return True
+    return False
+
+
+def cpg_ldlayout_map(ldlayout_dict):
+    if ldlayout_dict['RAIDType'] is not None and ldlayout_dict['RAIDType']:
+        ldlayout_dict['RAIDType'] = client.HPE3ParClient.RAID_MAP[
+            ldlayout_dict['RAIDType']]['raid_value']
+    if ldlayout_dict['HA'] is not None and ldlayout_dict['HA']:
+        ldlayout_dict['HA'] = getattr(
+            client.HPE3ParClient, ldlayout_dict['HA'])
+    return ldlayout_dict
+
+
+def create_cpg(
+        client_obj,
+        cpg_name,
+        domain,
+        growth_increment,
+        growth_limit,
+        growth_warning,
+        raid_type,
+        set_size,
+        high_availability,
+        disk_type,
+        new_name,
+        disable_auto_grow,
+        rm_growth_limit,
+        rm_warning_alert):
+    try:
+        if not validate_set_size(raid_type, set_size):
+            return (False, False, "Set size %s not part of RAID set %s" % (set_size, raid_type))
+        ld_layout = dict()
+        ld_layout_modify = dict()
+        modify_only_param = dict()
+        cpg_object_dict = dict()
+        modify_param_dict = dict()
+        disk_patterns = []
+        if disk_type:
+            disk_type = getattr(client.HPE3ParClient, disk_type)
+            disk_patterns = [{'diskType': disk_type}]
+        ld_layout = {
```

Comment:
Please update the dict by using the below syntax. what you are doing is re-initializing the variable and this creates a new id for the same member.

my_dict['my_key'] = 'my_value'
or
my_dict.update({'my_key': 'my_value'})

Do this in all other places where you are using a dict.